Flutter/lib/Server.dart

```diff
@@ -154,21 +154,24 @@ class Server {
     return responseSettle;
   }
 
-  static Future<Settle> submitVote(String option, [String code]) async {
+  static Future<Settle> submitVote(String option, bool done, [String code]) async {
     code = code ?? settleCode;
+    String finished = done == true ? "true" : "false";
 
     final http.Response response = await http.post(
       await _getUri(server_voting_path, code),
       headers: http_default_header,
       body: jsonEncode(<String, String>{

```

Comment:
on testing line 171 jsonDecode would sometimes crash, I don't know why but a solution that I found was to remove casting of <String, String> on this line (164)

Reply:
I believe the issue here might be issue #14 as the server returning an exception will also cause jsonDecode() to throw an exception. I decided to open an issue for it instead of fixing it now since we need to start wrapping up our winter break sprint in a few days and the issue is somewhat rare. Admittedly, we need to change the client to handle server exceptions better.